lib/player.js

```diff
@@ -8124,13 +8128,33 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           // Only update if we have key IDs for the stream.  If the keys aren't
           // all present, then the track should be restricted.
           if (stream.keyIds.size) {
-            variant.allowedByKeySystem = true;
-
-            for (const keyId of stream.keyIds) {
-              const keyStatus = keyStatusMap[isGlobalStatus ? '00' : keyId];
-              if (keyStatus || this.drmEngine_.hasManifestInitData()) {
-                variant.allowedByKeySystem = variant.allowedByKeySystem &&
-                    !!keyStatus && !restrictedStatuses.includes(keyStatus);
+            if (stream.drmInfos.length && !clearKeys.size) {
```

Comment:
Overall, I don't understand this change.  It's hard to read the logic on its own, and I don't get enough context from your PR description.

Maybe a refactor would help?  Or comments?

Reply:
I have added an explanatory comment with the use case.